### PR TITLE
Support Deploying MEANJS To Cloud Foundry

### PR DESCRIPTION
**Pull request by ryanjbaxter**
This is a port of #239 to the 0.4.0 branch.

I have tested the functionality against IBM Bluemix.  The only thing I did not successfully get to work is the login via PayPal.  I don't see exact instructions on how to do that yet so maybe my configuration is wrong.  If anyone has any advice on how to set that up correctly so I can test it that would be much appreciated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,6 +152,85 @@ $
 $ docker run -p 3000:3000 -p 35729:35729 -v /Users/mdl/workspace/mean-stack/mean/public:/home/mean/public -v /Users/mdl/workspace/mean-stack/mean/app:/home/mean/app --link db:db_1 mean
 ```
 
+## Deploying To Cloud Foundry
```
Comment (lirantal) [inline README.md]: I think up until line 188 the info is ok, but beyond it's quite specific to the CF configuration and that's not the place to put it. I suggest that we try one of:
1. We create a Wiki entry and refer to it
2. We create documentation for it on the 0.4.0 gh-pages branch and refer to it.

Also please place the CF README section in the bottom part and definitely not before the core MEANJS items like Getting Started area.

**Comment by lirantal** [comment]
Another note - I personally don't work with CF and I'm unable to test this - please make sure you are testing this and there aren't any issues with running it like this:
```
NODE_ENV=cloud-foundry grunt
```
To safely check that the correct environment is running.

File: README.md (hunk 1 above)
Comment (codydaig) [inline README.md]: Here's a barebone structure right now for the documentation:
https://github.com/meanjs/mean/blob/gh-pages/_includes/docs/0.4.x/overview.html

If you make another file with that structure with instructions on Deploying to CloudFoundry, I will get the link worked out in the documentation for it to show up.

**Comment by lirantal** [comment]
as @codydaig mentioned - he has done some good work on the 0.4.0 documentation work it out with him how to add it there.

and please squash your commits to something sane. we don't need all the cherrypick and other commits in the git history.

File: README.md (hunk 1 above)
Comment (ryanjbaxter) [inline README.md]: @codydaig is it best to create a separate pull request for that branch with the documentation in it?

**Comment by ryanjbaxter** [comment]
@lirantal I have tested it.  However anyone can easily test this (and I would appreciate it if someone would validate it as well).  You can signup for a free Bluemix account (based on CF) at bluemix.net.  You get a free 30 day trial and should be able to easily deploy it there following my instructions (assuming they are correct).

**Comment by lirantal** [comment]
Ok so:
@ryanjbaxter:
1. squash commits
2. work with @codydaig on a separate PR to update this information on the documentation area

**Comment by ryanjbaxter** [comment]
@lirantal I believe I squashed the commits correctly, let me know if not.

Waiting to hear back on @codydaig on how he would like me to add the documentation (and i will remove it from the README at that point)

**Comment by codydaig** [comment]
@ryanjbaxter I thought I commented already but it must be hidden in an outdated diff somewhere. 

Here's a barebone structure right now for the documentation:
https://github.com/meanjs/mean/blob/gh-pages/_includes/docs/0.4.x/overview.html

If you make another file with that structure with instructions on Deploying to CloudFoundry, I will get the link worked out in the other parts of the documentation for it to show up. (not all changes have merged yet on the main page so conflicts would arise)

**Comment by ryanjbaxter** [comment]
@codydaig yes got that, I just wanted to make sure you want me to submit a pull request against the gh-pages branch

**Comment by codydaig** [comment]
@ryanjbaxter Yes! Putting the file where the example I sent you is won't cause any merge issues. :-)

**Comment by lirantal** [comment]
@ryanjbaxter squash the commits so that we have just one, currently there are 8 and it's quite useless when we reflect back in git history. Thanks.

**Comment by ryanjbaxter** [comment]
@lirantal I have tried squashing the commits but seem to be getting into and endless look of merge conflicts at this point ( merged in changes made to the upstream 0.4 branch) not sure what i am doing wrong to be honest.  This is my first time squashing commits.

**Comment by lirantal** [comment]
@ryanjbaxter so to save you the trouble of handling the rebase/squash git thing just submit a new PR for this change and I'll close this and merge the new one.

**Comment by ilanbiala** [comment]
@ryanjbaxter is it possible for you to do this in the next day or so. We'd like to make this a part of the soon-to-be 0.4.0 release.

**Comment by ilanbiala** [comment]
Push to 0.4.x.

**Comment by ryanjbaxter** [comment]
@lirantal per your suggestion I created a new PR #732

**Pull request closed without merging**